src/main/java/bio/terra/service/resourcemanagement/DataLocationSelector.java

```diff
@@ -10,5 +10,5 @@
 
     String projectIdForFile(String datasetName, BillingProfileModel billingProfile);
 
-    String bucketForFile(String datasetName, BillingProfileModel billingProfile);
+    String bucketForFile(String datasetId, BillingProfileModel billingProfile);
```

Comment:
Would it be difficult to change this to accept `UUID` instead? It would be nice to avoid having the caller use `toString()` and would allow for slightly more precise type checking.